framework/src/base/EigenProblem.C

```diff
@@ -17,19 +17,29 @@
 #include "EigenProblem.h"
 #include "DisplacedProblem.h"
 #include "Assembly.h"
+#include "SlepcSupport.h"
+
+// libmesh includes
+#include "libmesh/system.h"
+#include "libmesh/eigen_solver.h"
 
 template<>
 InputParameters validParams<EigenProblem>()
 {
   InputParameters params = validParams<FEProblemBase>();
   params.addParam<unsigned int>("n_eigen_pairs", 1, "The dimension of the nullspace");
   params.addParam<unsigned int>("n_basis_vectors", 3, "The dimension of the nullspace");
```

Comment:
To me this parameter should be hidden to the user or at least group it into 'advanced'. Also give the recommendation on the value of this param in its doc string.